File: tests/fixtures/syntax/python/example_shapes.py
from abc import ABC, abstractmethod


class ExampleContract(ABC):
    @abstractmethod
    def example_required(self, example_value: str) -> str:
        raise NotImplementedError

    def example_concrete(self) -> str:
        return "example_value"


class ExampleImplementation(ExampleContract):
    example_optional: str | None

    def example_required(self, example_value: str) -> str:
        return example_value


async def example_async(example_value: str, *, example_suffix: str = "example_suffix") -> str:
    return example_value + example_suffix


def example_outer(example_value: str) -> str:
    def example_inner(example_input: str) -> str:
        return example_input

    return example_inner(example_value)

File: tests/fixtures/syntax/php/example_shapes.php
<?php
namespace ExampleShapes;

const EXAMPLE_CONSTANT = 'example_constant';
$example_variable = 'example_variable';

interface ExampleContract
{
    public function example_required(string $example_value): string;
}

abstract class ExampleAbstract
{
    abstract public function example_required(string $example_value): string;

    public function example_concrete(): string
    {
        return 'example_value';
    }
}

class ExampleImplementation extends ExampleAbstract implements ExampleContract
{
    public ?string $example_optional = null;

    public function example_required(string $example_value): string
    {
        return $example_value;
    }
}

function example_function(string $example_value, string $example_suffix = 'example_suffix'): string
{
    return $example_value . $example_suffix;
}
